jetty-server/src/test/java/org/eclipse/jetty/server/handler/InetAccessHandlerTest.java

```diff
@@ -0,0 +1,291 @@
+    private static InetAccessHandler _handler;
+
+    @BeforeAll
+    public static void setUp() throws Exception
+    {
+        _server = new Server();
+        _connector = new ServerConnector(_server);
+        _connector.setName("http");
+        _server.setConnectors(new Connector[]
+        { _connector });
+
+        _handler = new InetAccessHandler();
+        _handler.setHandler(new AbstractHandler()
+        {
+            @Override
+            public void handle(String target, Request baseRequest, HttpServletRequest request, HttpServletResponse response)
+                    throws IOException, ServletException
+            {
+                baseRequest.setHandled(true);
+                response.setStatus(HttpStatus.OK_200);
+            }
+        });
+        _server.setHandler(_handler);
+        _server.start();
+    }
+
+    /* ------------------------------------------------------------ */
+    @AfterAll
+    public static void tearDown() throws Exception
+    {
+        _server.stop();
+    }
+
+    /* ------------------------------------------------------------ */
+    @ParameterizedTest
+    @MethodSource("data")
+    public void testHandler(String include, String exclude, String includeConnectors, String excludeConnectors, String code)
+            throws Exception
+    {
+        _handler.clear();
+        for (String inc : include.split(";", -1))
+        {
+            if (inc.length() > 0)
+            {
+                _handler.include(inc);
+            }
+        }
+        for (String exc : exclude.split(";", -1))
+        {
+            if (exc.length() > 0)
+            {
+                _handler.exclude(exc);
+            }
+        }
+        for (String inc : includeConnectors.split(";", -1))
+        {
+            if (inc.length() > 0)
+            {
+                _handler.includeConnectorName(inc);
+            }
+        }
+        for (String exc : excludeConnectors.split(";", -1))
+        {
+            if (exc.length() > 0)
+            {
+                _handler.excludeConnectorName(exc);
+            }
+        }
+
+        String request = "GET /path HTTP/1.1\n" + "Host: 127.0.0.1\n\n";
+        Socket socket = new Socket("127.0.0.1", _connector.getLocalPort());
+        socket.setSoTimeout(5000);
+        try
+        {
+            OutputStream output = socket.getOutputStream();
+            BufferedReader input = new BufferedReader(new InputStreamReader(socket.getInputStream()));
+
+            output.write(request.getBytes(StandardCharsets.UTF_8));
+            output.flush();
+
+            Response response = readResponse(input);
+            Object[] params = new Object[]
+            { "Request WBHUC", include, exclude, includeConnectors, excludeConnectors, code, "Response", response.getCode() };
+            assertEquals(code, response.getCode(), Arrays.deepToString(params));
+        } finally
+        {
+            socket.close();
+        }
+    }
```

Comment:
Why cannot you use `HttpTester` here instead of doing manual HTTP/1.1 generation and parsing?
The `finally` is on the wrong line, so you may want to check your IDE settings.

Reply:
This test was copy/paste from IpAccessHandlerTest.  But sure - I'll rewrite